src/state/sandbox_exec.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/sh -ue
+
+POL='(version 1)(allow default)(deny network*)(deny file-write*)'
+POL="$POL"'(allow file-write* (literal "/dev/null"))'
+
+add_mounts() {
+    local DIR="$(cd "$2" && pwd -P)"
+    case "$1" in
+        ro) POL="$POL"'(deny file-write* (subpath "'"$DIR"'"))';;
+        rw) POL="$POL"'(allow file-write* (subpath "'"$DIR"'"))';;
+    esac
+}
+
+add_mounts rw "${TMPDIR:-/tmp}"
+
+# This case-switch should remain identical between the different sandbox implems
+COMMAND="$1"; shift
+case "$COMMAND" in
+    build)
+        add_mounts ro "$OPAM_SWITCH_PREFIX"
+        add_mounts rw "$PWD"
+        ;;
+    install)
+        add_mounts rw "$OPAM_SWITCH_PREFIX"
+        add_mounts ro "$OPAM_SWITCH_PREFIX/.opam-switch"
+        add_mounts rw "$PWD"
+        ;;
+    remove)
+        add_mounts rw "$OPAM_SWITCH_PREFIX"
+        add_mounts ro "$OPAM_SWITCH_PREFIX/.opam-switch"
+        [ "X${PWD#$OPAM_SWITCH_PREFIX/.opam-switch}" != "X${PWD}" ] && add_mounts rw "$PWD"
+        ;;
+    *)
+        echo "bwrap.sh: unknown command $COMMAND, must be one of 'build', 'install' or 'remove'" >&2
```

Comment:
this bwrap.sh could be $0 to be more accurate

Reply:
Indeed, thanks!